plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2822,7 +2822,10 @@ private ManagedObjectReference getVmfsDatastore(VmwareContext context, VmwareHyp
             throw new Exception("A relevant SCSI disk could not be located to use to create a datastore.");
         }
 
-        morDs = firstHostDatastoreSystemMO.createVmfsDatastore(datastoreName, hostScsiDisk);
+        morDs = firstHostDatastoreSystemMO.findDatastoreByName(datastoreName);
+        if (morDs == null) {
+            morDs = firstHostDatastoreSystemMO.createVmfsDatastore(datastoreName, hostScsiDisk);
```

Comment:
LGTM, the code is trying to find a DS by name and when it's not found it's creating a vmfs datastore. Should this be generic and enough to merge/accept this change - @harikrishna-patnala @shwstppr @nvazquez  @DaanHoogland ?